views/index.tpl

```diff
@@ -131,6 +131,11 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
     </p>
   </div>
 
+  <div>
+    <button type="button" onclick="codePermenateLinkGeneration()"> Generate link </button>
+    <input id = "codePermenateLink"> <p > </p>
```

Comment:
This doesn't look well-formed: where does the `input` element end? Should it be `<input id="..."/>`?
Spacing for `p` is strange.